add ca cert

marimlem opened the pull request:
# description:

creating our own Certificate Authority (CA) comes with extra steps, but at the advantage of not having to expose our backend to the public. amazing security step.

also you won't encounter the "untrusted cert, trust anyways" site again.

# downsides:

the downsides are the extra steps detailed in the README.md, strongly advise to read it. it is necessary for all devices entering our website to add our ca.crt manually to the trust-store of the system or the browser.

another downside, that i didn't mention in the readme, that the script and the steps in readme **might** be OS-dependent. I am working on ubuntu, and my VM at campus is also ubuntu, so chances are good it will work on campus. but for you testing, it might be even more extra steps, if your OS is acting different from ubuntu. 

# why do this?

the main reason for keeping the :3443 backend port exposed was to ensure that the websocket connection can be made. so far this is a manual step, you have to visit localhost:3443 in your browser and accept the certificate. but exposing the port, also exposes our API endpoints (and of course also our websocket endpoint). this is ... very bad, and something that has bothered me for a good while. but this PR solves it!

# potential sidesteps

in this PR I also deactivate the exposition of the backend ports :3000 and :3443. if you cannot make this work on your machine, or in the case of quick testing on other devices, like your phone or tablet, just _**expose :3443 again**_. this is also what i intend to do during evaluation, if we want to demonstrate playing a remote match from two different devices. you can do that by uncommenting line 9 in docker-compose.yml

<img width="407" height="223" alt="image" src="https://github.com/user-attachments/assets/eba4115e-6674-4110-aae7-8b20974daacb" />

# result

<img width="524" height="307" alt="image" src="https://github.com/user-attachments/assets/fbc22bbb-894a-4208-b136-dc270f9211b1" />

knusperpeter commented on backend/scripts/generate-ssl.sh:
```diff
@@ -14,46 +15,99 @@ fi
-    exit 1
-}
+# Get your computer's primary IPv4 address
+IP_ADDR=$(ip -4 addr show scope global | grep inet | awk '{print $2}' | cut -d/ -f1 | head -n1)
```
maybe check if this can fail to get the IP address.

marimlem commented:
# Nginx Reverse Proxy Implementation

## Overview
Implemented nginx as a reverse proxy to meet the `wss://` requirement while protecting the backend from direct public access.

## Key Changes

### 1. Added Nginx Configuration (`nginx.conf`)
```nginx
# Reverse proxy that:
# - Terminates SSL at the edge
# - Routes WebSocket upgrades properly
# - Protects backend from direct access
# - Handles Vite HMR WebSocket connections


**Key features:**

- SSL termination at nginx level
- WebSocket upgrade handling for both app and Vite HMR
- Selective proxying to internal services
- HTTP to HTTPS redirect
